src/Bridge/Symfony/Validator/Metadata/Property/Restriction/PropertySchemaRegexRestriction.php

```diff
@@ -29,7 +29,7 @@ class PropertySchemaRegexRestriction implements PropertySchemaRestrictionMetadat
      */
     public function create(Constraint $constraint, PropertyMetadata $propertyMetadata): array
     {
-        return isset($constraint->pattern) ? ['pattern' => $constraint->pattern] : [];
+        return $constraint->getHtmlPattern() ? ['pattern' => $constraint->getHtmlPattern()] : [];
```

Comment:
```suggestion
        return method_exists($constraint, 'getHtmlPattern') && $constraint->getHtmlPattern() ? ['pattern' => $constraint->getHtmlPattern()] : [];
```

(to prevent the PHPStan exception and bugs if the constraint doesn't implement this method).

Reply:
Solved with suggestion from @soyuka, see https://github.com/api-platform/core/pull/4027#discussion_r571302294